### PR TITLE
Remove pre cuda 11 code

### PR DESCRIPTION
**Pull request by MarcelKoch**
This PR remove all `#if` branches for CUDA < 11.

### CODE REVIEW

File: common/cuda_hip/components/atomic.hpp [hunk 1]
```diff
@@ -142,32 +143,26 @@ GKO_BIND_ATOMIC_ADD(double);
 #else  // NVIDIA
 
 
-#if !((defined(CUDA_VERSION) && (CUDA_VERSION < 8000)) || \
-      (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 600)))
+#if !(defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 600))
 // CUDA 8.0 starts suppoting 64-bit double atomicAdd on devices of compute
```
Comment (upsj) [inline common/cuda_hip/components/atomic.hpp:147]:
```suggestion
// CUDA supports 64-bit double atomicAdd on devices of compute
```

File: common/cuda_hip/components/atomic.hpp [hunk 2]
```diff
@@ -142,32 +143,26 @@ GKO_BIND_ATOMIC_ADD(double);
...
 GKO_BIND_ATOMIC_ADD(double);
-#endif  // !((defined(CUDA_VERSION) && (CUDA_VERSION < 8000)) ||
-        // (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 600)))
+#endif  // !(defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 600))
 
-#if !((defined(CUDA_VERSION) && (CUDA_VERSION < 10000)) || \
-      (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 700)))
+#if !(defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 700))
 // CUDA 10.0 starts supporting 16-bit __half floating-point atomicAdd on devices
 // of compute capability 7.x and higher.
```
Comment (upsj) [inline common/cuda_hip/components/atomic.hpp:154]:
```suggestion
// CUDA supports 16-bit __half floating-point atomicAdd on devices
// of compute capability 7.x and higher.
```

File: common/cuda_hip/components/atomic.hpp [hunk 3]
```diff
@@ -142,32 +143,26 @@ GKO_BIND_ATOMIC_ADD(double);
...
 GKO_BIND_ATOMIC_ADD(__half);
-#endif  // !((defined(CUDA_VERSION) && (CUDA_VERSION < 10000)) ||
-        // (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 700)))
+#endif  // !(defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 700))
 
-#if !((defined(CUDA_VERSION) && (CUDA_VERSION < 10000)) || \
-      (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 600)))
+#if !(defined(__CUDA_ARCH__) && (__CUDA_ARCH__ < 600))
 // CUDA 10.0 starts supporting 32-bit __half2 floating-point atomicAdd on
 // devices of compute capability 6.x and higher. note: The atomicity of the
```
Comment (upsj) [inline common/cuda_hip/components/atomic.hpp:160]:
```suggestion
// CUDA supports 32-bit __half2 floating-point atomicAdd on
// devices of compute capability 6.x and higher. note: The atomicity of the
```

File: cuda/base/cusparse_bindings.hpp [hunk 4]
```diff
@@ -886,7 +459,7 @@ inline void destroy(cusparseSpMatDescr_t descr)
 }
 
 
-#if (CUDA_VERSION >= 11031)
+#if defined(CUDA_VERSION) && (CUDA_VERSION >= 11031)
```
Comment (upsj) [inline cuda/base/cusparse_bindings.hpp:462]: Not sure if we even need these - the file should fail to compile if it's not compiled by nvcc or not including the necessary headers.

**Review by upsj — APPROVED** [review]
LGTM! We might even be able to remove some if not all `defined(CUDA_VERSION)` checks from the files.

**Review by yhmtsai — APPROVED** [review]

File: cuda/base/cusparse_bindings.hpp (hunk 4 above)
Comment (MarcelKoch) [reply cuda/base/cusparse_bindings.hpp:462]: I would just leave it, it's not hurting.

**Pull request merged**